mycore-validation/src/main/java/org/mycore/validation/pdfa/MCRPDFAFunctions.java

```diff
@@ -125,11 +130,77 @@ private static Document createXML(String objectId, Map<String, ValidationResult>
+            for (Element exception : exceptions) {
+                exceptionElements.appendChild(exception);
+            }
+            fileElement.appendChild(exceptionElements);
+        }
+    }
+    /**
+     * Creates a list of XML elements that represent the hierarchy of exceptions and their details.
+     * Each exception is represented as an XML element containing information about its class,
+     * message, and stack trace. The method processes the main throwable and any nested causes
+     * or suppressed exceptions.
+     *
+     * @param document  The XML {@link Document} to which the exception elements will belong.
+     *                  This document is used to create elements representing each exception.
+     * @param throwable The root {@link Throwable} to be processed. This throwable and its
+     *                  causal chain (including suppressed exceptions) are converted into
+     *                  XML elements.
+     * @return A list of XML {@link Element} objects, each representing a throwable in the
+     *         exception chain. Each element includes the exception class, message, and stack trace.
+     */
+
+    private static List<Element> createExceptionElements(Document document, Throwable throwable) {
+        List<Element> exceptionElementList = new ArrayList<>();
+        Queue<Throwable> exceptionsQueue = new LinkedList<>();
+        exceptionsQueue.add(throwable);
+
+        while (!exceptionsQueue.isEmpty()) {
+            Throwable currentException = exceptionsQueue.poll();
+            Element exceptionElement = document.createElement("exception");
+
+            Element classElement = document.createElement("class");
+            classElement.setAttribute("name", currentException.getClass().getName());
+            exceptionElement.appendChild(classElement);
+
+            if (currentException.getMessage() != null) {
+                Element messageElement = document.createElement("message");
+                messageElement.setAttribute("message", currentException.getMessage());
+                exceptionElement.appendChild(messageElement);
+            }
+
+            Element stackTraceElement = document.createElement("stackTrace");
+            for (StackTraceElement ste : currentException.getStackTrace()) {
+                Element frameElement = document.createElement("frame");
+                frameElement.setAttribute("text", ste.toString());
+                stackTraceElement.appendChild(frameElement);
+            }
+            exceptionElement.appendChild(stackTraceElement);
+
+            exceptionElementList.add(exceptionElement);
+
+            exceptionsQueue.addAll(List.of(currentException.getSuppressed()));
+            if (currentException.getCause() != null &&
+                !currentException.getCause().equals(currentException)) {
+                exceptionsQueue.add(currentException.getCause());
+            }
+        }
```

Comment:
This should be a hierarchy of elements not a list. We need to know what are causes and what a suppressed exception of which exception